src/core/actor/DipolarBarnesHut.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+  Copyright (C) 2016,2017,2018 The ESPResSo project
+
+  This file is part of ESPResSo.
+
+  ESPResSo is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  ESPResSo is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "config.hpp"
+#include "communication.hpp"
+#include "grid.hpp"
+#include "DipolarBarnesHut.hpp"
+#include "../forces.hpp"
+#include "EspressoSystemInterface.hpp"
+#include "forces.hpp"
+#include "energy.hpp"
+
+#ifdef DIPOLAR_BARNES_HUT
+
+void activate_dipolar_barnes_hut(float epssq, float itolsq)
+{
+    if (dipolarBarnesHut)
```

Comment:
This check is not necessary, cf. [[expr.delete]](http://eel.is/c++draft/expr.delete#2) of the C++ standard.

> [...] the value of the operand of delete may be a null pointer value [...]

You only have to make sure that *after* calling `delete` you set the value to `nullptr` so that calling `delete` twice will not crash the program.

Reply:
Thanks for the review. Done.